pkg/k8s/utils.go

```diff
@@ -12,3 +17,12 @@ func NamespacedName(obj metav1.Object) types.NamespacedName {
 		Name:      obj.GetName(),
 	}
 }
+
+// ServerVersion returns the version for k8s server
+func ServerVersion(client discovery.ServerVersionInterface) string {
+	versionInfo, err := client.ServerVersion()
+	if err != nil {
+		return unknownK8sVersion
```

Comment:
Would it help to log the error message ? I assume , this will be called only on restart.

Reply:
I would prefer we don't log error here since we are trying out best to get the k8s version here. If we can't get it shouldn't have any side impact on controller itself. So if we log error here might be a lot confusion to user which feels like there is something wrong with controller

Reply:
You can set a log level so that it's visible only in Debug mode, if thats the concern.But I am okay with this if its not critical, just a suggestion

Reply:
Thanks for the suggestion. I think we should be good here, if we need we can add it locally :)